tamr_unify_client/models/dataset/resource.py

```diff
@@ -94,6 +94,102 @@ def status(self) -> DatasetStatus:
             self.client, status_json, api_path=self.api_path + "/status"
         )
 
+    @property
+    def __geo_interface__(self):
+        """Retrieve a representation of this dataset that conforms to the Python Geo Interface.
+
+        Note that this materializes all features; for a streaming interface to features,
+        see :method:`~tamr_unify_client.models.dataset.Dataset.__geo_features__()`
+
+        See https://gist.github.com/sgillies/2217756
+
+        :return: dict[str, object]
+        """
+        return {
+            "type": "FeatureCollection",
+            "features": [feature for feature in self.iterfeatures()],
+        }
+
+    def iterfeatures(self):
+        """Returns an iterator that yields feature dictionaries that comply with __geo_interface__
+
+        See https://gist.github.com/sgillies/2217756
+
+        :return: stream of features
+        :rtype: Python generator yielding :py:class:`dict[str, object]`
+        """
+        key_attrs = self.key_attribute_names
+        if len(key_attrs) == 1:
+
+            def key_value(rec):
+                return rec[key_attrs[0]]
+
+        else:
+
+            def key_value(rec):
+                return [rec[attr] for attr in key_attrs]
+
+        # Duck-typing: find all the attributes that look like geometry
+        geo_attr_names = {
+            "point",
+            "multiPoint",
+            "lineString",
+            "multiLineString",
+            "polygon",
+            "multiPolygon",
+        }
+        geo_attrs = [
+            attr.name
+            for attr in self.attributes
+            if "RECORD" == attr.type.base_type
+            and geo_attr_names.intersection(
+                {sub_attr.name for sub_attr in attr.type.attributes}
+            )
+        ]
+        # We select the first such attribute as the geometry
+        if geo_attrs:
+            geo_attr = geo_attrs[0]
+        else:
+            geo_attr = None
```

Comment:
I guess if we wanted to store the other gemoetries as more than just simple properties, this is where we would do it:
```
for geo_attrs[1:]:
  <convert to something>
```